### PR TITLE
Add a SinkBinding sample.

### PR DESCRIPTION
**Pull request by mattmoor**
This is largely derived from the ContainerSource docs, but
using SinkBinding and a CronJob.

/hold

This needs the following related PRs to land:
 - https://github.com/knative/pkg/pull/915
 - https://github.com/knative/eventing/pull/2234
 - https://github.com/knative/eventing-contrib/pull/766

### CODE REVIEW

File: docs/eventing/samples/sinkbinding/README.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+---
+
+A SinkBinding is responsible for linking together "addressable" Kubernetes
+resources that may receive events (aka the event "sink") with Kubernetes
+resources that embed a PodSpec (as `spec.template.spec`) and want to produce
+events.
+
+The SinkBinding can be used to author new event sources using any of the
+familiar compute abstractions that Kubernetes makes available (e.g. Deployment,
+Job, DaemonSet, StatefulSet), or Knative abstractions (e.g. Service,
```
Comment (matzew) [inline docs/eventing/samples/sinkbinding/README.md]: add `CronJob`, since that is used in the sample ?

Comment (mattmoor) [reply docs/eventing/samples/sinkbinding/README.md]: CronJob actually isn't PodSpecable, we are binding the Jobs it creates.

**Comment by mattmoor** [comment]
All pre-reqs are in.
/hold cancel

File: docs/eventing/samples/sinkbinding/README.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+```yaml
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: event-display
+spec:
+  template:
+    spec:
+      containers:
+        - image: gcr.io/knative-releases/github.com/knative/eventing-sources/cmd/event_display
```
Comment (n3wscott) [inline docs/eventing/samples/sinkbinding/README.md]:
```suggestion
        - image: gcr.io/knative-releases/github.com/knative/eventing-contrib/cmd/event_display
```

Comment (mattmoor) [reply docs/eventing/samples/sinkbinding/README.md]: Pretty sure I copied this from the container source one, FYI

File: docs/eventing/samples/sinkbinding/sinkbinding.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+apiVersion: sources.eventing.knative.dev/v1alpha1
```
Comment (n3wscott) [inline docs/eventing/samples/sinkbinding/sinkbinding.yaml]: copyright

File: docs/eventing/samples/sinkbinding/heartbeats-source.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+apiVersion: batch/v1beta1
```
Comment (n3wscott) [inline docs/eventing/samples/sinkbinding/heartbeats-source.yaml]: copyright

**Comment by n3wscott** [comment]
/lgtm
/hold 
hold for copyrights

**Pull request merged**